elasticsearch/sgconfig/sg_config.yml

```diff
@@ -20,6 +20,9 @@ searchguard:
                prometheus:
                  verb: get
                  resource: /metrics
+               jaeger:
+                 verb: manage
```

Comment:
@jcantrill hi using the admin certs works fine with our operator. Now I have generated certs for jaeger and it seems that searchguard configuration is not right somewhere 

```
cat /var/run/secrets/kubernetes.io/serviceaccount/token 
eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJteXByb2plY3QiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIjoiamFlZ2VyLW9wZXJhdG9yLXRva2VuLTRtOHZ3Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQubmFtZSI6ImphZWdlci1vcGVyYXRvciIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6ImNjOGM1MDA5LTI2MDAtMTFlOS1hYzA2LThjMTY0NTZjODRlNyIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDpteXByb2plY3Q6amFlZ2VyLW9wZXJhdG9yIn0.dNqJfe4l1uI_Wm_bXCriIN0qWIfENmg2QkCHAyycvT0rOFOsNCcT4IX1u7fDm1jI2xHEddUO1tTlhMxGVlDjx4hjGqXdGxJx0X4kMlpC7dywTXOFeTnZ2ALntdkPl_RbSyUglhLX6qVaQ-5fV0sIWBPkRxVx3GyL4VS21vQOhBgiDnZex7U6zFdCkY6zoW1C7t6_bvUi6oKfkgBlAmni7Kql_mO0o2u7S8D3nmXk_6tFa0qKg02IKGWysJDiObOHRju0g8huOePM8fwJGTmhdntjTgLyb3mgKVuBBlUpY5Ar8_XP0FPH_CZyzRdtIg7hrG1yuEkKHdiAoV7QVDO_zA

curl -ivX GET -k  https://elasticsearch:9200/jaeger/_search -H "Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJteXByb2plY3QiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIjoiamFlZ2VyLW9wZXJhdG9yLXRva2VuLTRtOHZ3Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQubmFtZSI6ImphZWdlci1vcGVyYXRvciIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6ImNjOGM1MDA5LTI2MDAtMTFlOS1hYzA2LThjMTY0NTZjODRlNyIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDpteXByb2plY3Q6amFlZ2VyLW9wZXJhdG9yIn0.dNqJfe4l1uI_Wm_bXCriIN0qWIfENmg2QkCHAyycvT0rOFOsNCcT4IX1u7fDm1jI2xHEddUO1tTlhMxGVlDjx4hjGqXdGxJx0X4kMlpC7dywTXOFeTnZ2ALntdkPl_RbSyUglhLX6qVaQ-5fV0sIWBPkRxVx3GyL4VS21vQOhBgiDnZex7U6zFdCkY6zoW1C7t6_bvUi6oKfkgBlAmni7Kql_mO0o2u7S8D3nmXk_6tFa0qKg02IKGWysJDiObOHRju0g8huOePM8fwJGTmhdntjTgLyb3mgKVuBBlUpY5Ar8_XP0FPH_CZyzRdtIg7hrG1yuEkKHdiAoV7QVDO_zA"

{"error":{"root_cause":[{"type":"security_exception","reason":"no permissions for [indices:data/read/search] and User [name=system:serviceaccount:myproject:jaeger-operator, roles=[gen_kibana_c53e044ccf54e1ca0f0d90ce80e0e083f4bc2fec, gen_user_c53e044ccf54e1ca0f0d90ce80e0e083f4bc2fec]]"}],"type":"security_exception","reason":"no permissions for [indices:data/read/search] and User [name=system:serviceaccount:myproject:jaeger-operator, roles=[gen_kibana_c53e044ccf54e1ca0f0d90ce80e0e083f4bc2fec, gen_user_c53e044ccf54e1ca0f0d90ce80e0e083f4bc2fec]]"}
```

Is this verb ok? Or should it represent HTTP method?

Reply:
I believe the verb is what ever you define for the Openshift auth resource.  It's considered virtual and doesn't really correspond to any actual type in the system.  Looks like CRUD and MANAGE dont actually correspond to SEARCH. You might find this script [2] useful which will dump the permissions for the cluster.  

[1] https://github.com/openshift/origin-aggregated-logging/blob/master/elasticsearch/sgconfig/sg_action_groups.yml#L50
[2] https://github.com/jcantrill/cluster-logging-tools/blob/master/scripts/view-es-permissions

Reply:
Should jaeger reuse one of those defined in `sg_action_groups.yml`? 

What is should be if we need to write and read to a set of jaeger indices?